File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    int result;
    public int goodNodes(TreeNode root) {
        result = 0;
        dfs(root, -101);
        return result;
    }

    public void dfs(TreeNode node, int max) {
        if (node == null) return;
        int newMax = max;
        if (node.val >= max) {
            newMax = node.val;
            result++;
        }

        dfs(node.left, newMax);
        dfs(node.right, newMax);
    }
}

File: Data Structures & Algorithms/islands-and-treasure/submission-0.py
class Solution:
    def islandsAndTreasure(self, grid: List[List[int]]) -> None:
        def dfs(d, i, j):
            if (i < 0 or 
                i >= len(grid) or 
                j < 0 or 
                j >= len(grid[0]) or
                grid[i][j] < d):
                return
            
            grid[i][j] = d
            dfs(d + 1, i + 1, j)
            dfs(d + 1, i - 1, j)
            dfs(d + 1, i, j + 1)
            dfs(d + 1, i, j - 1)
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == 0: dfs(0, i, j)